validator/sawtooth_validator/execution/scheduler_serial.py

```diff
@@ -42,54 +42,64 @@ def __init__(self, squash_handler, first_state_hash):
         self._final = False
         self._complete = False
         self._cancelled = False
+        self._last_context_id = None
+        self._last_valid_batch_c_id = None
         self._squash = squash_handler
         self._condition = Condition()
         # contains all txn.signatures where txn is
         # last in it's associated batch
         self._last_in_batch = []
-        self._last_state_hash = first_state_hash
+        self._previous_state_hash = first_state_hash
+        # The state hashes here are the ones added in add_batch, and
+        # are the state hashes that correspond with block boundaries.
+        self._required_state_hashes = {}
+        self._always_persist = always_persist
 
     def __iter__(self):
         return SchedulerIterator(self, self._condition)
 
     def set_transaction_execution_result(
             self, txn_signature, is_valid, context_id):
-        """the control flow is that on every valid txn a new state root is
-        generated. If the txn is invalid the batch status is set,
-        if the txn is the last txn in the batch, is valid, and no
-         prior txn failed the batch, the
-        batch is valid
-        """
         with self._condition:
             if (self._in_progress_transaction is None or
                     self._in_progress_transaction != txn_signature):
-                raise ValueError("transaction not in progress: {}",
-                                 txn_signature)
+                raise ValueError("transaction not in progress: {}".format(
+                                 txn_signature))
             self._in_progress_transaction = None
 
             if txn_signature not in self._txn_to_batch:
                 raise ValueError("transaction not in any batches: {}".format(
                     txn_signature))
+
+            batch_signature = self._txn_to_batch[txn_signature]
             if is_valid:
-                # txn is valid, get a new state hash
-                state_hash = self._squash(self._last_state_hash, [context_id])
-                self._last_state_hash = state_hash
+                self._last_context_id = context_id
```

Comment:
Is this the previous context id?  If so, we should use `_previous_context_id`, to differentiate from `_last_in_batch`: the two together make it seem like the final one in a list.